src/archive_converter.py

```diff
@@ -21,55 +21,44 @@
 from message_dao import MessageDao
 
 class ArchiveMessageIndex(object):
-    def __init__(self, message_dao : MessageDao) -> None:
+    def __init__(self, message_dao: MessageDao) -> None:
         self._message_dao = message_dao
 
-    def update(self, data_dir: str) -> List[Message]:
+    def update(self, data_dir: str) -> Dict[str, Message]:
         """ Updates index with messages in the passed in directory.
-        Returns a list of new messages """
+        Returns a dictionary mapping new messages' ids to their corresponding message."""
 
-        new_messages : List[Message] = []
+        new_messages : Dict[str, Message] = {}
 
         for filename in os.listdir(data_dir):
             if not filename.endswith(".txt"):
                 continue
             email = generate_email_from_file(os.path.join(data_dir, filename))
             if email and not self._message_dao.get(email.id):
-                new_messages.append(email)
+                new_messages[email.id] = email
         self._add_messages_to_index(new_messages)
         return new_messages
 
-    def size(self) -> int:
-        return self._message_dao.size()
-
-    def find(self, message_id : str) -> Message:
-        message = self._message_dao.get(message_id)
-        if message is None:
-            raise ValueError(f'Could not find message: {message_id}')
-        return message
-
-    def _add_messages_to_index(self, new_messages : List[Message]) -> None:
+    def _add_messages_to_index(self, new_messages: Dict[str, Message]) -> None:
```

Comment:
nit: can we rename this function now?
the name implies we might be mutating the current object by storing the new messages.

Reply:
Renamed method to indicate that the purpose is to add children as needed.